diff --git a/src/OpenColorIO/FileRules.cpp b/src/OpenColorIO/FileRules.cpp
index 794dfdbe..20a4301b 100644
--- a/src/OpenColorIO/FileRules.cpp
+++ b/src/OpenColorIO/FileRules.cpp
@@ -6,6 +6,7 @@
 #include <map>
 #include <regex>
 #include <sstream>
+#include <cstring>
 
 #include <OpenColorIO/OpenColorIO.h>
 
@@ -61,7 +62,7 @@ std::string ConvertToRegularExpression(const char * globPattern, bool ignoreCase
 
     if (ignoreCase)
     {
-        const size_t length = strlen(globPattern);
+        const size_t length = std::strlen(globPattern);
         bool respectCase = false;
         for (size_t i = 0; i < length; ++i)
         {
